### PR TITLE
Remove a lot of unnecessary Java type parameters

### PR DESCRIPTION
**Pull request by snazy**
Generics removed from:
* `StoreWorker`, uses `Content`, `Content.Type`, `CommitMeta` directly
* `VersionStore`, uses `Content`, `Content.Type`, `CommitMeta` directly
* `Commit`, uses `Content`, `CommitMeta` directly
* `Operation` + `Delete`/`Put`/`Unchanged`, uses `Content` directly
* `Diff`, uses `Content` directly
* `KeyEntry`, uses `Content.Type` directly

Changes to `Content` + `Content.Type`:
* `Content.Type` refactored to a "plain" Java interface
* Add ability to register `Content.Type`s via Java service-loader
  mechanism
* Updated custom `Content`/`Content.Type`/`CommitMeta` variants
  in `nessie-versioned-tests`, add a content-type bundle for these

Note: Weld/CDI extension for compatibility tests had to be updated
to allow (old) parameterized-type signature of `VersionStore` _and_
(new) non-parameterized-type signature.

This PR is split into two commits: one with the actual changes and another with the "boring stuff/changes" implied by the actual changes.

Follow-up PRs: #4942 #4943 #4944 #4945

Noteworthy changes in the follow-up PRs:
* `Content.Type.payload()` will be removed in a follow-up PR
* Many uses of `StoreWorker` will be eliminated

See issue #4940

### CODE REVIEW

**Comment by snazy** [comment]
Quarkus doesn't handle `@JsonTypeIdResolver` yet (https://github.com/quarkusio/quarkus/issues/27346). - had to add it to `ModelReflections` for now.

File: model/src/main/java/org/projectnessie/model/types/DefaultContentTypeBundle.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import org.projectnessie.model.IcebergTable;
+import org.projectnessie.model.IcebergView;
+import org.projectnessie.model.Namespace;
+import org.projectnessie.model.types.ContentTypes.Register;
+
+/**
+ * Provides the {@link org.projectnessie.model.Content.Type content types} for Iceberg table + view,
+ * Delta Lake table and namespaces.
+ */
+public final class DefaultContentTypeBundle implements ContentTypeBundle {
```
Comment (dimas-b) [inline model/src/main/java/org/projectnessie/model/types/DefaultContentTypeBundle.java:28]: rename to `MainContentTypeBundle` maybe? "Default" implies that we'd use it if there's nothing else, but all bundles are combined on load, right?

File: versioned/tests/src/main/java/org/projectnessie/versioned/testworker/MockContentTypeBundle.java [hunk 2]
```diff
@@ -15,18 +15,15 @@
...
 
-  enum Type {
-    /** Content type with on-reference state. */
-    ON_REF_ONLY,
-    /** Content type with on-reference state and mandatory global state. */
-    WITH_GLOBAL_STATE,
-    /** Content type with on-reference state and content attachments. */
-    WITH_ATTACHMENTS
-  }
+public final class MockContentTypeBundle implements ContentTypeBundle {
```
Comment (dimas-b) [inline versioned/tests/src/main/java/org/projectnessie/versioned/testworker/MockContentTypeBundle.java:21]: `TestContentTypeBundle`? These are real type implementations :thinking:

File: versioned/persist/adapter/build.gradle.kts [hunk 3]
```diff
@@ -30,7 +30,9 @@ dependencies {
   compileOnly(platform(project(":nessie-deps-build-only")))
   annotationProcessor(platform(project(":nessie-deps-build-only")))
 
+  implementation(project(":nessie-model"))
   implementation(project(":nessie-versioned-spi"))
+  compileOnly("org.eclipse.microprofile.openapi:microprofile-openapi-api")
```
Comment (dimas-b) [inline versioned/persist/adapter/build.gradle.kts:35]: Why? I might have missed the changes that need the new dependencies... :thinking:

**Review by dimas-b — COMMENTED** [review]
LGTM overall. Minor comments.

File: gc/gc-base/src/main/java/org/projectnessie/gc/base/ContentBloomFilter.java [hunk 4]
```diff
@@ -72,13 +72,12 @@ public boolean wasMerged() {
...
 
   private String getValue(Content content) {
-    switch (content.getType()) {
-      case ICEBERG_TABLE:
-        return ICEBERG_TABLE.name() + ((IcebergTable) content).getSnapshotId();
-      case ICEBERG_VIEW:
-        return ICEBERG_VIEW.name() + ((IcebergView) content).getVersionId();
-      default:
-        throw new RuntimeException("Unsupported type " + content.getType());
+    if (content.getType().equals(ICEBERG_TABLE)) {
```
[the remark this answers is not in the document]
Comment (snazy) [reply gc/gc-base/src/main/java/org/projectnessie/gc/base/ContentBloomFilter.java:75]: Not an enum

File: versioned/persist/adapter/build.gradle.kts (hunk 3 above)
Comment (snazy) [reply versioned/persist/adapter/build.gradle.kts:35]: Annotations leaking into the compilation (otherwise there are compilation warnings)

Comment (snazy) [reply versioned/persist/adapter/build.gradle.kts:35]: Also: `Content` leaks here as well.

File: servers/quarkus-common/src/main/java/org/projectnessie/quarkus/reflect/ModelReflections.java [hunk 5]
```diff
@@ -25,12 +26,14 @@
...
+@SuppressWarnings("unused")
 @RegisterForReflection(
     targets = {
       NessieError.class,
       CELUtil.KeyedEntityForCel.class,
       CELUtil.OperationForCel.class,
       CELUtil.ContentForCel.class,
-      CELUtil.KeyEntryForCel.class
+      CELUtil.KeyEntryForCel.class,
+      ContentTypeIdResolver.class
```
[the remark this answers is not in the document]
Comment (snazy) [reply servers/quarkus-common/src/main/java/org/projectnessie/quarkus/reflect/ModelReflections.java:37]: Technically it's a Quarkus bug ;)

**Review by dimas-b — APPROVED** [review]

**Pull request merged**